js/chmod_test.ts

```diff
@@ -0,0 +1,147 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import { testPerm, assertEqual } from "./test_util.ts";
+import * as deno from "deno";
+
+const isNotWindows = deno.platform.os !== "win";
+
+testPerm({ write: true }, function chmodSyncSuccess() {
+  const enc = new TextEncoder();
+  const data = enc.encode("Hello");
+  const tempDir = deno.makeTempDirSync();
+  const filename = tempDir + "/test.txt";
+  deno.writeFileSync(filename, data, 0o666);
+
+  let fileInfo = deno.statSync(filename);
+  assertEqual(fileInfo.mode & 0o777, 0o666);
+
+  // On windows, no effect, but should not crash
+  deno.chmodSync(filename, 0o777);
+
+  // Check success when not on windows
+  if (isNotWindows) {
+    fileInfo = deno.statSync(filename);
+    assertEqual(fileInfo.mode & 0o777, 0o777);
+  }
+});
+
+// Check symlink when not on windows
+if (isNotWindows) {
+  testPerm({ write: true }, function chmodSyncSymlinkSuccess() {
+    const enc = new TextEncoder();
+    const data = enc.encode("Hello");
+    const tempDir = deno.makeTempDirSync();
+
+    const filename = tempDir + "/test.txt";
+    deno.writeFileSync(filename, data, 0o666);
+    const symlinkName = tempDir + "/test_symlink.txt";
+    deno.symlink(filename, symlinkName);
+
+    let fileInfo = deno.statSync(filename);
+    assertEqual(fileInfo.mode & 0o777, 0o666);
+    let symlinkInfo = deno.lstatSync(symlinkName);
+    const symlinkMode = symlinkInfo.mode & 0o777; // plaform dependent
+
+    deno.chmodSync(symlinkName, 0o777);
+
+    // Change actual file mode, not symlink
+    fileInfo = deno.statSync(filename);
+    assertEqual(fileInfo.mode & 0o777, 0o777);
+    symlinkInfo = deno.lstatSync(symlinkName);
+    assertEqual(symlinkInfo.mode & 0o777, symlinkMode);
+  });
+}
+
+testPerm({ write: true }, function chmodSyncFailure() {
+  let err;
+  try {
+    const filename = "/badfile.txt";
+    deno.chmodSync(filename, 0o777);
+  } catch (e) {
+    err = e;
+  }
+  assertEqual(err.kind, deno.ErrorKind.NotFound);
+  assertEqual(err.name, "NotFound");
+});
+
+testPerm({ write: false }, function chmodSyncPerm() {
+  let err;
+  try {
+    const filename = deno.makeTempDirSync() + "/test.txt";
```

Comment:
I think `deno.makeTempDirSync()` raise a PermissionDenied error.

Reply:
Ooops. Thanks for pointing out